### DISCUSSION TITLE
Accept common FlatZinc search annotations even when they are not supported

### DISCUSSION THREAD

**Discussion by zayenz** [google/or-tools#3343] [Feature requests] [answered]
**What language and solver does this apply to?**
FlatZinc interface. Tested both on latest release (9.3) and latest main (78939381c2).

**Describe the problem you are trying to solve.**
Running a single MiniZinc model on many different solvers is complicated by the fact that OR-Tools crashes/errors hard on unsupported search annotations. Consider the following model
```
int: n = 2;
array[1..n] of var 1..n: x;

solve :: int_search(x, dom_w_deg, indomain_min) 
satisfy;
```
that uses the quite common annotation [dom_w_deg](https://www.minizinc.org/doc-2.6.3/en/lib-stdlib-annotations.html?highlight=dom_w_deg#mzn-stdlib-annotations-search-varsel-dom-w-deg) from the standard MiniZinc library. Running the above model gives the error
```
F0613 16:31:23.059969     1 cp_model_fz_solver.cc:1038] Unsupported order: dom_w_deg
*** Check failure stack trace: ***
    @        0x10d546b88  google::LogMessage::SendToLog()
    @        0x10d5472ad  google::LogMessage::Flush()
    @        0x10d54a77f  google::LogMessageFatal::~LogMessageFatal()
    @        0x10d547cc9  google::LogMessageFatal::~LogMessageFatal()
    @        0x10b5e00bf  operations_research::sat::SolveFzWithCpModelProto()
    @        0x102ee8e83  main
    @        0x202f9251e  start
    @                0x5  0x0
=====ERROR=====
Process finished with non-zero exit code 1.
Finished in 148msec.
```

**Describe the solution you'd like**
Even when a non-supported search annotation is used, I think it would be better to warn instead of crash hard.

**Describe alternatives you've considered**
The problem can be scripted around, but it is a source of annoyance, especially since it is currently not possible to make MiniZinc models that have any type of conditional compilation based on the solver. Solutions are either to use different model files or to supply the annotation as data.

**Comment by lperron** [comment, marked as answer]
Fixed on the main branch.

Thanks

**Comment by Mizux** [comment]
For the record: https://github.com/google/or-tools/commit/2115d65c35181366dd80733eac979d3bcfa53146

note: github seems to not perform auto cross link for discussion....
